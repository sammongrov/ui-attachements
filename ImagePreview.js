import React, { Component } from 'react';
import {
  View,
  Image,
  KeyboardAvoidingView,
  Platform,
  TextInput,
  TouchableOpacity,
} from 'react-native';
import { styles } from 'react-native-theme';
import { Actions } from 'react-native-router-flux';
import PropTypes from 'prop-types';
import { Colors } from '@ui/theme_default';
import { NavBar, Icon } from '@ui/components';

export default class ImagePreview extends Component {
  constructor(props) {
    super(props);
    this.imageData = this.props.imageData; // eslint-disable-line
    this.imageUrl = this.props.imageUrl; // eslint-disable-line
    this.state = {
      messageText: '',
      imageUrl: this.imageUrl,
      height: 40,
    };
    this.groupDetail = this.props.groupDetail; // eslint-disable-line
    this.onSuccessAction = this.props.onSuccessAction; // eslint-disable-line
    this.PreviewSuccess = this.PreviewSuccess.bind(this);
  }

  componentWillMount() {
    // console.log('data-viswa', this.imageData);
  }

  updateSize = (height) => {
    this.setState({
      height,
    });
  };

  PreviewSuccess() {
    const { messageText } = this.state;
    const { onSuccessAction } = this.props;
    onSuccessAction(messageText);
    // this.onSuccessAction(this.groupDetail, this.imageData, messageText);
  }

  render() {
    const { imageUrl, messageText, height } = this.state;
    const newStyle = { height };
    return (
      <View style={[styles.flex1, styles.blackBackground]}>
        <NavBar
          leftComponent={
            <TouchableOpacity
              style={[styles.navSideButtonDimension, styles.alignJustifyCenter]}
              onPress={() => {
                if (Actions.currentScene === 'ImagePreviewScene') {
                  Actions.pop();
                }
              }}
            >
              <Icon
                name="chevron-left"
                type="material-community"
                color={Colors.NAV_ICON}
                size={36}
              />
            </TouchableOpacity>
          }
          titleText="Image Preview"
        />
        <Image
          style={styles.alignCenterJustifyEnd}
          source={{
            uri: imageUrl,
          }}
          resizeMode="contain"
        />
        <KeyboardAvoidingView
          behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
          style={[styles.imagePreviewOverlay, styles.imagePreviewKeyboardView]}
        >
          <View style={[styles.rowFlex, styles.imagePreviewMessagecontainer]}>
            <TextInput
              placeholder="Add Caption..."
              style={[styles.flex1, styles.imagePreviewTextInput, newStyle]}
              onChangeText={(text) => {
                this.setState({ messageText: text });
              }}
              value={messageText}
              underlineColorAndroid="transparent"
              multiline={true}
              onContentSizeChange={(e) => this.updateSize(e.nativeEvent.contentSize.height)}
            />
            <TouchableOpacity
              style={[styles.alignJustifyCenter, styles.imagePreviewSendButton]}
              onPress={() => {
                if (Actions.currentScene === 'ImagePreviewScene') {
                  this.PreviewSuccess();
                }
              }}
            >
              <Icon name="send" size={20} color={Colors.ICON_WHITE} />
            </TouchableOpacity>
          </View>
        </KeyboardAvoidingView>
      </View>
    );
  }
}

ImagePreview.propTypes = {
  imageUrl: PropTypes.string,
  imageData: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  groupDetail: PropTypes.object,
  onSuccessAction: PropTypes.func,
};

ImagePreview.defaultProps = {
  imageUrl: '',
  imageData: {},
  groupDetail: {},
  onSuccessAction: () => {},
};
