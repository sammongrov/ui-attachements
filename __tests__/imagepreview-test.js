import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Actions } from 'react-native-router-flux';
import ImagePreview from '../ImagePreview';

configure({ adapter: new Adapter() });
const onSuccessAction = jest.fn();
const imageUrl = 'https://my-photo-library.com/image-1.png';
const imageData = { width: 320, height: 160, caption: 'coffee' };
const groupDetail = { _id: 'GENERAL', title: 'All about coffee' };

jest.mock('react-native-router-flux', () => ({
  Actions: {
    currentScene: 'ImagePreviewScene',
    pop: jest.fn(),
  },
}));

jest.mock('Platform', () => {
  const Platform = require.requireActual('Platform');
  Platform.OS = 'android';
  return Platform;
});

it('renders correctly with props', () => {
  const tree = renderer
    .create(
      <ImagePreview
        onSuccessAction={onSuccessAction}
        imageUrl={imageUrl}
        imageData={imageData}
        groupDetail={groupDetail}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly without props', () => {
  const tree = renderer.create(<ImagePreview />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('onPress on Profile', () => {
  const tree = shallow(<ImagePreview />).dive();
  const image = tree.find('TouchableOpacity');
  image.props().onPress();
  expect(image.props().onPress).toBeInstanceOf(Function);
});

it('updateSize', () => {
  const rootComponent = shallow(<ImagePreview />);
  const instance = rootComponent.instance();
  const result = instance.updateSize();
  expect(result).toBeUndefined();
});

it('PreviewSuccess is called', () => {
  const rootComponent = shallow(<ImagePreview onSuccessAction={onSuccessAction} />);
  const instance = rootComponent.instance();
  const result = instance.PreviewSuccess();
  expect(result).toBeUndefined();
  expect(onSuccessAction.mock.calls.length).toBe(1);
});

it('PreviewSuccess is not called', () => {
  onSuccessAction.mockClear();
  Actions.currentScene = 'ChatList';
  const root = shallow(<ImagePreview onSuccessAction={onSuccessAction} />);
  root
    .find('TouchableOpacity')
    .last()
    .props()
    .onPress();
  expect(onSuccessAction.mock.calls.length).toBe(0);
});

it('should render the onChange to messageText', () => {
  const rootComponent = shallow(<ImagePreview />);
  const authorInputComponent = rootComponent.find('TextInput').first();
  authorInputComponent.simulate('ChangeText', 'text');
  expect(rootComponent.state('messageText')).toEqual('text');
});

it('calls onContentSizeChange', () => {
  const event = { nativeEvent: { contentSize: { height: 150 } } };
  const rootComponent = shallow(<ImagePreview />);
  const input = rootComponent.find('TextInput').first();
  input.props().onContentSizeChange(event);
  expect(rootComponent.state('height')).toEqual(150);
});

it('Navbar onPress calls Actions.pop', () => {
  jest.doMock('Platform', () => {
    const Platform = require.requireActual('Platform');
    Platform.OS = 'ios';
    return Platform;
  });
  Actions.currentScene = 'ImagePreviewScene';
  const root = shallow(<ImagePreview />);
  const button = root
    .find('NavBar')
    .shallow()
    .find('TouchableOpacity');
  button.props().onPress();
  expect(Actions.pop.mock.calls.length).toBe(1);
});

it('Navbar onPress does not calls Actions.pop', () => {
  Actions.currentScene = 'ChatList';
  Actions.pop.mockClear();
  const root = shallow(<ImagePreview />);
  const button = root
    .find('NavBar')
    .shallow()
    .find('TouchableOpacity');
  button.props().onPress();
  expect(Actions.pop.mock.calls.length).toBe(0);
});
