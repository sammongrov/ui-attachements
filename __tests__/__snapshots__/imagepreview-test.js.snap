// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with props 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        null,
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#2E88FF",
                "fontSize": 36,
              },
              undefined,
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          type="material-community"
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          undefined,
          undefined,
          null,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "OpenSans-Regular",
            },
            Array [
              undefined,
              null,
            ],
          ]
        }
      >
        Image Preview
      </Text>
    </View>
    <View
      style={
        Array [
          undefined,
          undefined,
        ]
      }
    />
  </View>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "https://my-photo-library.com/image-1.png",
      }
    }
  />
  <View
    enabled={true}
    keyboardVerticalOffset={0}
    onLayout={[Function]}
    style={
      Array [
        Array [
          undefined,
          undefined,
        ],
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          undefined,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        multiline={true}
        onChangeText={[Function]}
        onContentSizeChange={[Function]}
        placeholder="Add Caption..."
        style={
          Array [
            undefined,
            undefined,
            Object {
              "height": 40,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly without props 1`] = `
<View
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        undefined,
        null,
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#2E88FF",
                "fontSize": 36,
              },
              undefined,
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          type="material-community"
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          undefined,
          undefined,
          null,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "OpenSans-Regular",
            },
            Array [
              undefined,
              null,
            ],
          ]
        }
      >
        Image Preview
      </Text>
    </View>
    <View
      style={
        Array [
          undefined,
          undefined,
        ]
      }
    />
  </View>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "",
      }
    }
  />
  <View
    enabled={true}
    keyboardVerticalOffset={0}
    onLayout={[Function]}
    style={
      Array [
        Array [
          undefined,
          undefined,
        ],
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
          undefined,
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        multiline={true}
        onChangeText={[Function]}
        onContentSizeChange={[Function]}
        placeholder="Add Caption..."
        style={
          Array [
            undefined,
            undefined,
            Object {
              "height": 40,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
